lib/controllers/git-tab-controller.js

```diff
@@ -327,16 +333,25 @@ export default class GitTabController {
     return this.getActiveRepository().checkout(branchName, options);
   }
 
-  focus() {
-    this.refs.gitTab.focus();
+  @autobind
+  rememberLastFocus(event) {
+    this.lastFocus = this.refs.gitTab.rememberFocus(event) || GitTabView.focus.STAGING;
   }
 
-  focusAndSelectStagingItem(filePath, stagingStatus) {
-    return this.refs.gitTab.focusAndSelectStagingItem(filePath, stagingStatus);
+  restoreFocus() {
+    this.refs.gitTab.setFocus(this.lastFocus);
+  }
+
+  hasFocus() {
+    return this.element.contains(document.activeElement);
```

Comment:
Ah, no I was thinking of a method that bit me once because it *wasn't* inclusive... wonder what I'm thinking of?